lib/puppet/provider/file_line/ssh.rb

```diff
@@ -37,7 +37,7 @@ def handle_create_with_match()
       raise Puppet::Error, "More than one line in file '#{resource[:path]}' matches pattern '#{resource[:match]}'"
     elsif match_count == 1
       Puppet.debug('Replacing config line')
-      transport.exec!("sed -i \"s/#{resource[:match]}.*/#{resource[:line]}/\" #{resource[:path]}")
+      transport.exec!("sed -i 's|#{resource[:match]}.*|#{resource[:line]}|' #{resource[:path]}")
```

Comment:
No. The sed utility allows you to use any non-alpha or special character as a delimiter. so instead of using the "/" character which then has to be escaped (e.g. https:\/\/~) using the "|" as the command delimiter eliminates the need for the escapes.